plan/planbuilder.go

```diff
@@ -325,6 +329,88 @@ func (b *planBuilder) buildAdmin(as *ast.AdminStmt) Plan {
 	return p
 }
 
+func detachIndexColumns(tn *ast.TableName) (indexOffset []int, columnOffset []int) {
```

Comment:
Eh, index columns will use more effective algo, normal columns also have histograms.
Also, I think that your example needs only three histograms, because histogram for columns `a` can be inferred from index `(a,b)`.